agent/api/task_unix.go

```diff
@@ -25,6 +38,70 @@ func getCanonicalPath(path string) string { return path }
 
 // CgroupEnabled checks whether cgroups need be enabled at the task level
 func (task *Task) CgroupEnabled() bool {
-	//stub for now
+	// TODO: check for more conditions
+	// CPU + Mem -> true
+	// CPU -> true
+	// Mem -> false
+	// !CPU || !Mem -> false
+	if !utils.ZeroOrNil(task.VCPULimit) {
+		return true
+	}
 	return false
 }
+
+// BuildCgroupRoot helps build the task cgroup prefix
+// Example: /ecs/task-id
+func (task *Task) BuildCgroupRoot() (string, error) {
+	taskID, err := task.GetID()
+	if err != nil {
+		return "", errors.Wrapf(err, "task build cgroup root: unable to get task-id")
+	}
+
+	cgroupRoot := strings.Join([]string{config.DefaultTaskCgroupPrefix, taskID}, sepForwardSlash)
+	return cgroupRoot, nil
+}
+
+// BuildLinuxResourceSpec returns a linuxResources object for the task cgroup
+func (task *Task) BuildLinuxResourceSpec() specs.LinuxResources {
+	linuxResourceSpec := specs.LinuxResources{}
+
+	if !(utils.ZeroOrNil(task.VCPULimit)) {
```

Comment:
Will this task be started with `cpu limit` zero in this case? Should this return an error if the `task.VCPULimit` is zero? The same for the `memory limit`.

Reply:
The `taskDef` validation happens elsewhere. 
This method would be called only when task cgroups are required.